subprojects/docs/src/docs/userguide/jvm/building_java_projects.adoc

```diff
@@ -224,21 +224,36 @@ include::sample[dir="snippets/java/basic/kotlin",files="build.gradle.kts[tags=ja
 
 That's also how you can change the verbosity of the compiler, disable debug output in the byte code and configure where the compiler can find annotation processors.
 
-Two common options for the Java compiler are defined at the project level:
+[[sec:java_cross_compilation]]
+=== Targeting a specific Java version
+
+By default, Gradle will compile Java code to the language level of the JVM running Gradle.
+
+Since version 9, the Java compiler can be configured to produce bytecode for an older Java version, additionally making sure the code does not use any APIs from a more recent version.
+Gradle now supports this link:{groovyDslPath}/org.gradle.api.tasks.compile.CompileOptions.html#org.gradle.api.tasks.compile.CompileOptions:release[release] flag on `CompileOptions` directly for Java compilation.
+This option takes precedence over the properties described below.
+
+Historical options for the Java compiler remain available, at the project level, on the `java` extension:
 
 `sourceCompatibility`::
 Defines which language version of Java your source files should be treated as.
 
 `targetCompatibility`::
 Defines the minimum JVM version your code should run on, i.e. it determines the version of byte code the compiler generates.
 
-If you need or want more than one compilation task for any reason, you can either <<#sec:custom_java_source_sets,create a new source set>> or simply define a new task of type link:{groovyDslPath}/org.gradle.api.tasks.compile.JavaCompile.html[JavaCompile]. We look at setting up a new source set next.
+Alternatively, these options can be set per link:{groovyDslPath}/org.gradle.api.tasks.compile.JavaCompile.html[JavaCompile] task, using properties with the same names.
 
-[[sec:java_cross_compilation]]
-=== Compiling and testing Java 6/7
+However these options do not protect against the use of APIs introduced in later Java versions.
+
+==== Compiling and testing Java 6/7
 
 Gradle can only run on Java version 8 or higher.
 
+[NOTE]
```

Comment:
This note feels out of place here after only the sentence above. The sentence above and below could use a rewrite to clarify that this is needed if you actually want to run an older Java executable. And that you can use "release" if it is just about the version...

Or you just move this note below the next sentence. :) (Which could be combined into one paragraph with the sentence above).

Reply:
Moved indeed